ketchup/watch_schedule_builder.py

```diff
@@ -45,4 +48,11 @@ def calculate_watch_schedule(episodes, end_date, start_date, per_day_limit=None)
             # It isn't possible to watch with that limit so...
             response["is_possible"] = False
 
+    if per_week_limit:
```

Comment:
So lets write an additional test (in the separate test file) that would effect this log.

Consider something like testing the math itself... a test case could be like:

Total episodes -> 15
Episodes Per Day limit -> 5 eps
Days per week limit -> 3 days per week
total eps per week -> 15 episodes per week
start date -> August 21, 2023
end date -> September 30, 2023

assert that math is correct  (only 1 week is required to watch the show)

Then do a different test where one thing changes:

**Total episodes -> 150**
Episodes Per Day limit -> 5 eps
Days per week limit -> 3 days per week
total eps per week -> 15 episodes per week
start date -> August 21, 2023
end date -> September 30, 2023

we know this case isnt possible in that time frame, so we should assert the response has changed.
You can assert on multiple things (this is a contentious topic in the industry, you can see some of that [here in this article](https://methodpoet.com/are-multiple-asserts-bad/). )   So we can assert :

```
response = {
        "base_eps_per_day": 0,
        "days": 0,
        "actual_watch_days": 0,
        "is_possible": True,
        "realistic_end_date": None,
        "extra_episodes": None,
    }
```

Each of these values are set as we expect. This way, the user who calls this function can get the response they'd expect. So you could assert that response['base_eps_per_day'] is equal to x, same with days, etc.

Now you can start to consider multiple cases for this function ! 

Ideally, we want every line of this function tested, and the best way to do that is to activate those if statements via different input variables. Does this make sense?